### DISCUSSION TITLE
Treat a property as nullable

### DISCUSSION THREAD

**Discussion by jamesfoster** [riok/mapperly#302] [Q&A] [answered]
I need a way to treat a property as nullable even tho the source type declares it as non-nullable. The problem is I don't own the source type and in certain cases this "non-nullable" property is coming out as null 🙄.

Any help appreciated.

**Comment by latonz** [comment, marked as answer]
This is currently not possible with Mapperly. As a workaround you can exclude the property via the `MapperIgnoreTarget`/`MapperIgnoreSource ` attributes and map it manually.

**Comment by jamesfoster** [comment]
Thanks, I've also noticed that when mapping a nullable struct to the same nullable struct it generates the following:

`target.Prop = source.Prop ?? default;`

This is interpreted as:

`target.Prop = source.Prop ?? default(SomeStruct);`

instead of:

`target.Prop = source.Prop ?? default(Nullable<SomeStruct>);` (or indeed just `target.Prop = source.Prop`)

So instead of `null` mapping to `null`, `null` actually maps to the default instance of the struct. In the case of a `DateTime`, null becomes `1970-01-01T00:00:00Z`

**Reply by latonz** [reply]
This should be fixed in the latest pre-release (`2.8.0-next.1`), could you verify this?

**Reply by jamesfoster** [reply]
Yes, it's now correct 👍.

I updated the nuget reference to that version and re-built and it still came out as the same `?? default`.

I had to restart VS to get it to use the new version it seems. Is that expected behaviour?

**Reply by latonz** [reply]
I think sometimes NuGet packages get cached but I'm not aware of the details, usually a version change of a package should also download the new package and load the new source generator.
